### DISCUSSION TITLE
.buffer() while keeping original shape

### DISCUSSION THREAD

**Discussion by dorianherle** [shapely/shapely#2044] [Q&A]
My question is also posted here: 
https://stackoverflow.com/questions/78187566/how-to-offset-a-shaply-polygon-without-chnaging-corner-shape

I have the issue that I cannot buffer a shaply polygon such that the corner/shape stays the same. Here is my issue. Compare the buffered and original polygon:

![image](https://github.com/shapely/shapely/assets/32427425/6818867b-9255-45a4-9f02-32add8df9b2c)

It should keep the rounded corners...

![image](https://github.com/shapely/shapely/assets/32427425/c567eb09-84af-492c-9767-9120430bffc9)

Here is the code

```
import matplotlib.pyplot as plt
from shapely.geometry import Polygon
import geopandas as gpd
from matplotlib.lines import Line2D  # Import for custom legend

# original polygon definition
original_poly = Polygon([(0, 0), (2, 1), (5, 1), (5, 2), (2, 2), (0, 2)])

# Apply dilation then erosion to round the corners
original_poly = original_poly.buffer(0.25).buffer(-0.5).buffer(0.25)

# compute offset polygon
offset_poy = original_poly.buffer(0.5, resolution=16, join_style=1)

## PLOT

# Create GeoSeries from the polygons
original_poly_gseries = gpd.GeoSeries([original_poly])
offseted_poly_gseries = gpd.GeoSeries([offseted_polygon])

# Plotting
fig, ax = plt.subplots()
# Improved color palette and adjusted alpha for better visualization
original_poly_gseries.plot(ax=ax, color='#1f77b4', alpha=0.7)  # A nice shade of blue
offseted_poly_gseries.plot(ax=ax, color='k', alpha=0.7)  # A complementary shade of orange

# Creating custom legends with updated colors
legend_elements = [
    Line2D([0], [0], color='#1f77b4', lw=4, label='Original Polygon'),
    Line2D([0], [0], color='k', lw=4, label='Offset Polygon'),
]

# Placing the legend outside the plot
ax.legend(handles=legend_elements, loc='upper left', bbox_to_anchor=(1, 1))
plt.show()
```

**Comment by jorisvandenbossche** [comment]
I cannot reproduce your output. I assume you meant to take a "negative" offset and the  `buffer(0.5, ..)` should be `buffer(-0.5, ..)`?

```python
from shapely.geometry import Polygon

# original polygon definition
original_poly = Polygon([(0, 0), (2, 1), (5, 1), (5, 2), (2, 2), (0, 2)])

# Apply dilation then erosion to round the corners
original_poly = original_poly.buffer(0.25).buffer(-0.5).buffer(0.25)

# compute offset polygon
offset_poly = original_poly.buffer(-0.5, resolution=16, join_style=1)

# plot result
from shapely.plotting import plot_polygon
plot_polygon(original_poly, add_points=False, alpha=0.7)
plot_polygon(offset_poly, add_points=False, color="k", alpha=0.7)
```

that now gives

![Figure_1](https://github.com/shapely/shapely/assets/1020496/29e46117-23cf-4609-bd8e-7b528119e7af)

To your original question about preserving the rounding: by default, this _is_ actually preserved. You can notice that when reducing the buffer distance, for example from -0.5 to -0.15 gives the figure below:

![Figure_1](https://github.com/shapely/shapely/assets/1020496/8a7578fc-6060-40ba-a3a8-3a60f2257713)

It's just that with the buffer distance of 0.5, this rounding is almost not noticeable anymore because of the erosion (and the requirement to keep the distance to the original border equal)
